volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,646 @@
+            )
+
+    @classmethod
+    def parse_inventory_app_key(
+        cls, inv_app_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, _AmcacheEntry]]:
+        """Parses InventoryApplication entries from the Windows 10 `Root\\InventoryApplication` key.
+
+        :param programs_key: The `Root\\InventoryApplication` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\InventoryApplicationFile` entries, and the second member is the `AmcacheEntry`.
+        """
+        for program_key in inv_app_key.get_subkeys():
+            program_id = program_key.get_name()
+
+            values = {
+                str(value.get_name()): value
+                for value in program_key.get_values()
+                if value.get_name() in [key.value for key in Win10InvAppValName]
+            }
+
+            name = _get_string_value(values, Win10InvAppValName.Name.value)
+            version = _get_string_value(values, Win10InvAppValName.Version.value)
+            publisher = _get_string_value(values, Win10InvAppValName.Publisher.value)
+            path = _get_string_value(values, Win10InvAppValName.RootDirPath.value)
+            install_date = _get_datetime_str_value(
+                values, Win10InvAppValName.InstallDate.value
+            )
+            last_mod = conversion.wintime_to_datetime(
+                program_key.LastWriteTime.QuadPart
+            )
+
+            product: str = name if isinstance(name, str) else "UNKNOWN"  # type: ignore
+            if isinstance(version, str):
+                product += " " + version
+
+            yield program_id.strip().strip("\u0000"), _AmcacheEntry(
+                AmcacheEntryType.Program.name,
+                path=path,
+                last_modify_time=last_mod,
+                install_time=install_date,
+                product_name=product,
+                company=publisher,
+            )
+
+    @classmethod
+    def parse_inventory_app_file_key(
+        cls, inv_app_file_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[NullableString, _AmcacheEntry]]:
+        """Parses executable file entries from the `Root\\InventoryApplicationFile` registry key.
+
+        :param inv_app_file_key: The `Root\\InventoryApplicationFile` registry key.
+        :return: An iterator of tuples, where the first member is the program ID string for correlating
+        with it's parent `InventoryApplication` program entry, and the second member is the `Amcache` entry.
+        """
+
+        valName = Win10InvAppFileValName
+
+        for file_key in inv_app_file_key.get_subkeys():
+
+            vollog.debug(
+                f"Parsing Win10 InventoryApplicationFile key {file_key.get_name()}"
+            )
+
+            values = {
+                str(value.get_name()): value
+                for value in file_key.get_values()
+                if value.get_name() in [key.value for key in valName]
```

Comment:
Again, this list comprehension could be done before the loop even starts rather than every run of a loop inside a loop...